### PR TITLE
fix for mysql 8.0

### PR DESCRIPTION
**Pull request by jinzhen-lin**
my_bool is no longer supported in mysql 8.0. https://github.com/mysql/mysql-server/commit/eb821c023cedc029ca0b06456dfae365106bee84#diff-7adc547ff2f20b0b8a32d74e644a757fL171

### CODE REVIEW

File: src/pch.h [hunk 1]
```diff
@@ -4,3 +4,7 @@
 #include <plogr.h>
 
 using namespace Rcpp;
+
+#if MYSQL_VERSION_ID >= 80000 && MYSQL_VERSION_ID < 100000
```
Comment (krlmlr) [inline src/pch.h]: What does it mean if `MYSQL_VERSION_ID < 100000` ?

**Comment by jinzhen-lin** [comment]
I find that the fails of test were caused by the `DBItest` package on CRAN. 

On my computer(OS: Manjaro Linux, R 3.5.1, MySQL 8.0), if i install `DBItest` from CRAN, then the test would fail, but if i use `devtools::install_github("r-dbi/DBItest")` to reinstall `DBItest`, then it can pass all tests.

**Comment by krlmlr** [comment]
Thanks for the heads-up, I restarted the Travis build with clean caches. Could you please rebase to current master and force-push?

**Comment by jinzhen-lin** [comment]
do you mean I should remove the commit that I use for testing travis? If so, I have done this before.

File: src/pch.h (hunk 1 above)
Comment (krlmlr) [reply src/pch.h]: So is this patch also valid when building against Connector-C for MariaDB?

Comment (jinzhen-lin) [reply src/pch.h]: The patch does nothing except the change from `std::vector` to `boost::container::vector` if MariaDB is installed. MariaDB have not remove `my_bool` so far, and without `MYSQL_VERSION_ID < 100000`, the package cannot be built. But it is OK if the condition is added. I have tested that with MariaDB 10.1 and MariaDB 10.3 on Manjaro.

Comment (jinzhen-lin) [reply src/pch.h]: But now I wonder if MariaDB would remove `my_bool` in the future. Maybe it is better to use `#if !defined(my_bool)` ?

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**